pkg/localstore/localstore.go

```diff
@@ -204,7 +205,7 @@ type Options struct {
 // New returns a new DB.  All fields and indexes are initialized
 // and possible conflicts with schema from existing database is checked.
 // One goroutine for writing batches is created.
-func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger logging.Logger) (db *DB, err error) {
+func New(ctx context.Context, path string, baseKey []byte, ss storage.StateStorer, o *Options, logger logging.Logger) (db *DB, err error) {
```

Comment:
Not sure about this. This would then mean that the Localstore would then have to adhere to context semantics. This is not the case. We have `db.Close` which should be used to close the localstore. So for `collectGarbage` you can use the background context or a cancellable context which is canceled on localstore.Close.